src/services/kuma-api/RestClient.ts

```diff
@@ -1,27 +1,25 @@
 import { makeRequest } from './makeRequest'
+import type { EnvVars } from '@/services/env/Env'
 
-const DEFAULT_OPTIONS: RequestInit = {
-  credentials: 'include',
-}
+type Env = (
+  key: keyof Pick<EnvVars,
+  'KUMA_API_URL'
+  >
+) => string
```

Comment:
Because it means that we don't need `Env` (and fill all that out) just to test. I vaguely remember I made some notes on this in the PR where I originally added it. I was very in two minds about making things loosely coupled but ensuring typescript auto complete. I'd rather just go with loose coupling tbh, but I think the idea here was to keep a balance.

Probably a longer story here but that's the TLDR

Reply:
Wouldn't it make more sense then to add a type assertion in the test instead or altering the source code?

Reply:
It means you don't need `Env` at all in the tests you can just `() => ''`. If you use Env you have to fill out all its values which are many. I much prefer the looser coupling.

Reply:
I just tried this and you don't even need the type assertion:

```ts
import type Env from '../env/Env'
```

and

```
  constructor(
    protected env: Env['var'],
  ) {
    this._baseUrl = env('KUMA_API_URL')
  }
```

works.

Reply:
I changed this, I couldn't repro the issue I found a few months back and it looks like I didn't note it in the original PR so 🤷